src/map-marker/floating-marker.js

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright (c) Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import * as React from 'react';
+import {useStyletron} from '../styles/index.js';
+import PinHead from './pin-head.js';
+import {
+  StyledFloatingMarkerRoot,
+  StyledFloatingMarkerAnchorContainer,
+  StyledFloatingMarkerPinHeadContainer,
+} from './styled-components.js';
+import type {FloatingMarkerPropsT} from './types.js';
+import {
+  FLOATING_MARKER_ANCHOR_POSITIONS,
+  PINHEAD_SIZES,
+  PINHEAD_TYPES,
+  FLOATING_MARKER_ANCHOR_TYPES,
+  anchorSize,
+} from './constants.js';
+
+const FloatingMarker = ({
+  color,
+  background,
+  label,
+  size = PINHEAD_SIZES.medium,
+  anchor = FLOATING_MARKER_ANCHOR_POSITIONS.bottomLeft,
+  endEnhancer,
+  startEnhancer,
+  anchorType = FLOATING_MARKER_ANCHOR_TYPES.circle,
+}: FloatingMarkerPropsT) => {
+  const [, theme] = useStyletron();
+  const {
+    colors: {backgroundPrimary, backgroundInversePrimary, primaryA, primaryB},
+  } = theme;
+  color = color || primaryA;
+  background = background || backgroundPrimary;
+
+  const anchorPinHeadSize =
+    anchorType === FLOATING_MARKER_ANCHOR_TYPES.circle
+      ? PINHEAD_SIZES.xSmallCircle
+      : PINHEAD_SIZES.xSmallSquare;
```

Comment:
It's a bit confusing that `PINHEAD_SIZES` also includes shapes in the enum. Maybe create a new enum called `PINHEAD_SHAPE`. I see that only x-small allows for different shapes. Maybe if another size calls for square shape, log a warning instructing the developer that it will not take an effect

Reply:
@khudiono-uber and I were discussing this, but this seems like it would bring unnecessary complexity to the API. I know it fall under bad practices to implement like this, but are you okay with keeping pin head sizes to include circle and square for xsmall?

Reply:
This was the updated version to split to size + shape if we want to go back to it: https://github.com/uber/baseweb/pull/4548/commits/29443673cc5262eea56ddde94274183c264a2afd#diff-2af6f172c72c7a37e39d9fa8ed04beeeab2e72758d9db2c1d2afa4bd113b62d3R61

I don't necessarily think it adds complexity to the API. I'll defer to the base web folks whether or not they want to keep it the way it is now or with the split size/shape

Reply:
I fine with either splitting to two enums or generalizing the enum title from PINHEAD_SIZES to something that can describe size + shape